### ISSUE TITLE
[achecker command]: Can't run a folder with subfolders

### ISSUE THREAD

**Issue by gdelory** [IBMa/equal-access#171]
## What tool(s) are you using?

- [ ] `accessibility-checker-engine`
- [x] `accessibility-checker`
- [ ] `karma-accessibility-checker`
- [ ] `rule-server`
- [ ] `boilerplates`
- [ ] `other`

## Detailed description

> Node version: 10
> Platform: Windows/Linux

> Describe in detail the issue you're having.

## Steps to reproduce the issue

1. Create a `testfolder` folder with a few sub-folders and a few html files in `testfolder` and each subfolders
2. run the command `npx achecker testfolder`

## Additional information

Hi, so it took me a really long time to figure out why I couldn't use achecker on my folder with subfolders, but when running this command:

`npx achecker src`

I'm getting the following error:

```
TypeError: Cannot convert undefined or null to object
    at Function.e.clearCaches (/dcs/accessibility/node_modules/accessibility-checker/lib/engine/ace-node.js:16:97784)
    at e.run (/dcs/accessibility/node_modules/accessibility-checker/lib/engine/ace-node.js:16:98062)
    at e.check (/dcs/accessibility/node_modules/accessibility-checker/lib/engine/ace-node.js:16:95147)
    at Object.aChecker.getComplianceHelperLocal (/dcs/accessibility/node_modules/accessibility-checker/lib/ACHelper.js:656:40)
    at Object.aChecker.getComplianceHelper (/dcs/accessibility/node_modules/accessibility-checker/lib/ACHelper.js:476:35)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:2796) UnhandledPromiseRejectionWarning: TypeError: Cannot convert undefined or null to object
    at Function.e.clearCaches (/dcs/accessibility/node_modules/accessibility-checker/lib/engine/ace-node.js:16:97784)
    at e.run (/dcs/accessibility/node_modules/accessibility-checker/lib/engine/ace-node.js:16:98062)
    at e.check (/dcs/accessibility/node_modules/accessibility-checker/lib/engine/ace-node.js:16:95147)
    at Object.aChecker.getComplianceHelperLocal (/dcs/accessibility/node_modules/accessibility-checker/lib/ACHelper.js:656:40)
    at Object.aChecker.getComplianceHelper (/dcs/accessibility/node_modules/accessibility-checker/lib/ACHelper.js:476:35)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:2796) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:2796) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

I started by enabling DEBUG with a `export DEBUG=true`, but didn't give me so much more:

```
acc@3de9c9c1154d:/dcs/accessibility$ npx achecker src
START 'processConfiguration' function
START 'loadConfigFromYAMLorJSONFile' function
Working directory set to: /dcs/accessibility
Checking file: /dcs/accessibility/.achecker.yml
File: /dcs/accessibility/.achecker.yml exists loading it.
Loading as YAML file.
Loaded config from file:
{ ruleArchive: '03June2020',
  policies: [ 'IBM_Accessibility' ],
  failLevels: [ 'violation' ],
  reportLevels:
   [ 'violation',
     'potentialviolation',
     'recommendation',
     'manual',
     'potentialrecommendation' ],
  outputFormat: [ 'json', 'html' ],
  label: [ 'master' ],
  outputFolder: 'results',
  extensions: [ 'html', 'htm', 'svg' ],
  headless: true }
Using config which was loaded from config file.
START 'initializeDefaults' function
Config before initialization:
{ ruleArchive: '03June2020',
  policies: [ 'IBM_Accessibility' ],
  failLevels: [ 'violation' ],
  reportLevels:
   [ 'violation',
     'potentialviolation',
     'recommendation',
     'manual',
     'potentialrecommendation' ],
  outputFormat: [ 'json', 'html' ],
  label: [ 'master' ],
  outputFolder: 'results',
  extensions: [ 'html', 'htm', 'svg' ],
  headless: true }
Config after initialization:
{ ruleArchive: '03June2020',
  policies: [ 'IBM_Accessibility' ],
  failLevels: [ 'violation' ],
  reportLevels:
   [ 'violation',
     'potentialviolation',
     'recommendation',
     'manual',
     'potentialrecommendation' ],
  outputFormat: [ 'json', 'html' ],
  label: [ 'master' ],
  outputFolder: 'results',
  extensions: [ 'html', 'htm', 'svg' ],
  headless: true,
  rulePack: 'https://able.ibm.com/rules/archives/latest/js',
  baselineFolder: undefined,
  checkHiddenContent: false,
  engineFileName: 'ace.js',
  captureScreenshots: false,
  toolID: 'accessibility-checker-v3.0.6',
  scanID: 'd64a534e-8b16-4151-9100-8d8d3e42e92b' }
END 'initializeDefaults' function
START 'processACConfig' function
START 'convertPolicies' function
Converting policy provided to Array:
[ 'IBM_Accessibility' ]
END 'convertPolicies' function
Found archiveFile: https://able.ibm.com/rules/archives.json
Built new rulePack: https://able.ibm.com/rules/archives/2020.06.03/js
END 'processACConfig' function
END 'processConfiguration' function
START ACReporter Constructor
END ACReporter Constructor
START ACReporter Constructor
END ACReporter Constructor
START ACReporter Constructor
END ACReporter Constructor
START 'aChecker.getCompliance' function
[INFO] aChecker.loadEngine detected Puppeteer
START 'aChecker.isLabelUnique' function
Checking if label: /dcs/accessibility/src/README.html is unique.
END 'aChecker.isLabelUnique' function
getComplianceHelper:Puppeteer
sendResultsToReporter: [ 'json', 'html' ]
START 'info' emitter function
START 'savePageResults' function
Results are going to be stored under results directory: "results"
START 'writeObjectToFileAsJSON' function
Parent Directoy: "results/dcs/accessibility/src"
Object will be written to file: "results/dcs/accessibility/src/README.html.json"
END 'writeObjectToFileAsJSON' function
END 'savePageResults' function
START 'addResultsToGlobal' function
Adding following object to scanSummary.pageScanSummary:
{ label: '/dcs/accessibility/src/README.html',
  counts:
   { violation: 100,
     potentialviolation: 3,
     recommendation: 0,
     manual: 0,
     potentialrecommendation: 0,
     ignored: 0 } }
END 'addResultsToGlobal' function
END 'info' emitter function
START 'info' emitter function
START 'savePageResults' function
Results are going to be stored under results directory: "results"
START 'writeObjectToFileAsHTML' function
Parent Directoy: "results/dcs/accessibility/src"
Object will be written to file: "results/dcs/accessibility/src/README.html.html"
END 'writeObjectToFileAsHTML' function
END 'savePageResults' function
START 'addResultsToGlobal' function
Adding following object to scanSummary.pageScanSummary:
{ label: '/dcs/accessibility/src/README.html',
  counts: { ignored: 0 } }
END 'addResultsToGlobal' function
END 'info' emitter function
START 'profileV2' function
END 'profileV2' function
Failed: /dcs/accessibility/src/README.html
START 'aChecker.getCompliance' function
[INFO] aChecker.loadEngine detected Puppeteer
START 'aChecker.isLabelUnique' function
Checking if label: /dcs/accessibility/src/Release_notes.html is unique.
END 'aChecker.isLabelUnique' function
getComplianceHelper:Puppeteer
sendResultsToReporter: [ 'json', 'html' ]
START 'info' emitter function
START 'savePageResults' function
Results are going to be stored under results directory: "results"
START 'writeObjectToFileAsJSON' function
Parent Directoy: "results/dcs/accessibility/src"
Object will be written to file: "results/dcs/accessibility/src/Release_notes.html.json"
END 'writeObjectToFileAsJSON' function
END 'savePageResults' function
START 'addResultsToGlobal' function
Adding following object to scanSummary.pageScanSummary:
{ label: '/dcs/accessibility/src/Release_notes.html',
  counts:
   { violation: 7,
     potentialviolation: 1,
     recommendation: 0,
     manual: 0,
     potentialrecommendation: 0,
     ignored: 0 } }
END 'addResultsToGlobal' function
END 'info' emitter function
START 'info' emitter function
START 'savePageResults' function
Results are going to be stored under results directory: "results"
START 'writeObjectToFileAsHTML' function
Parent Directoy: "results/dcs/accessibility/src"
Object will be written to file: "results/dcs/accessibility/src/Release_notes.html.html"
END 'writeObjectToFileAsHTML' function
END 'savePageResults' function
START 'addResultsToGlobal' function
Adding following object to scanSummary.pageScanSummary:
{ label: '/dcs/accessibility/src/Release_notes.html',
  counts: { ignored: 0 } }
END 'addResultsToGlobal' function
END 'info' emitter function
START 'profileV2' function
END 'profileV2' function
Failed: /dcs/accessibility/src/Release_notes.html
START 'aChecker.getCompliance' function
[INFO] aChecker.loadEngine detected Puppeteer
START 'aChecker.isLabelUnique' function
Checking if label: /dcs/accessibility/src/SUMMARY.html is unique.
END 'aChecker.isLabelUnique' function
getComplianceHelper:Puppeteer
sendResultsToReporter: [ 'json', 'html' ]
START 'info' emitter function
START 'savePageResults' function
Results are going to be stored under results directory: "results"
START 'writeObjectToFileAsJSON' function
Parent Directoy: "results/dcs/accessibility/src"
Object will be written to file: "results/dcs/accessibility/src/SUMMARY.html.json"
END 'writeObjectToFileAsJSON' function
END 'savePageResults' function
START 'addResultsToGlobal' function
Adding following object to scanSummary.pageScanSummary:
{ label: '/dcs/accessibility/src/SUMMARY.html',
  counts:
   { violation: 646,
     potentialviolation: 0,
     recommendation: 0,
     manual: 0,
     potentialrecommendation: 0,
     ignored: 0 } }
END 'addResultsToGlobal' function
END 'info' emitter function
START 'info' emitter function
START 'savePageResults' function
Results are going to be stored under results directory: "results"
START 'writeObjectToFileAsHTML' function
Parent Directoy: "results/dcs/accessibility/src"
Object will be written to file: "results/dcs/accessibility/src/SUMMARY.html.html"
END 'writeObjectToFileAsHTML' function
END 'savePageResults' function
START 'addResultsToGlobal' function
Adding following object to scanSummary.pageScanSummary:
{ label: '/dcs/accessibility/src/SUMMARY.html',
  counts: { ignored: 0 } }
END 'addResultsToGlobal' function
END 'info' emitter function
START 'profileV2' function
END 'profileV2' function
Failed: /dcs/accessibility/src/SUMMARY.html
START 'aChecker.getCompliance' function
[INFO] aChecker.loadEngine detected local
START 'aChecker.isLabelUnique' function
Checking if label: [object Promise] is unique.
END 'aChecker.isLabelUnique' function
getComplianceHelper:Local
TypeError: Cannot convert undefined or null to object
    at Function.e.clearCaches (/dcs/accessibility/node_modules/accessibility-checker/lib/engine/ace-node.js:16:97784)
    at e.run (/dcs/accessibility/node_modules/accessibility-checker/lib/engine/ace-node.js:16:98062)
    at e.check (/dcs/accessibility/node_modules/accessibility-checker/lib/engine/ace-node.js:16:95147)
    at Object.aChecker.getComplianceHelperLocal (/dcs/accessibility/node_modules/accessibility-checker/lib/ACHelper.js:656:40)
    at Object.aChecker.getComplianceHelper (/dcs/accessibility/node_modules/accessibility-checker/lib/ACHelper.js:476:35)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:2861) UnhandledPromiseRejectionWarning: TypeError: Cannot convert undefined or null to object
    at Function.e.clearCaches (/dcs/accessibility/node_modules/accessibility-checker/lib/engine/ace-node.js:16:97784)
    at e.run (/dcs/accessibility/node_modules/accessibility-checker/lib/engine/ace-node.js:16:98062)
    at e.check (/dcs/accessibility/node_modules/accessibility-checker/lib/engine/ace-node.js:16:95147)
    at Object.aChecker.getComplianceHelperLocal (/dcs/accessibility/node_modules/accessibility-checker/lib/ACHelper.js:656:40)
    at Object.aChecker.getComplianceHelper (/dcs/accessibility/node_modules/accessibility-checker/lib/ACHelper.js:476:35)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:2861) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:2861) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
^Cacc@3de9c9c1154d:/dcs/accessibility$
```

So I debugged a bit more by adding my own log in various places, and noticed that some objects where Promises of paths, which looked really odd. Following this up the the bin/achecker.js, it seems like `getInputFileList` do not build the list of files properly, and concat Promises instead of the actual result:
```
rptInputFiles: [ 'src/README.html',
  'src/Release_notes.html',
  'src/SUMMARY.html',
  Promise {
    [ 'src/about/accessibility.html',
      'src/about/cloud_paks.html',
      'src/about/deprecated.html',
      'src/about/disable_net_isolation.html',
      'src/about/known_issues.html',
      'src/about/landing.html',
      'src/about/languages.html',
      'src/about/notices.html',
      'src/about/support.html',
      'src/about/whats_new.html' ] },
  Promise {
    [ [Promise],
      [Promise],
      'src/audit-logging/changelog-audit-logging.html',
      'src/audit-logging/landing_audit-logging.html' ] },
  Promise { [ [Promise], 'src/calico-cli/changelog-calico.html' ] },
  Promise {
    [ [Promise],
      [Promise],
      'src/catalog/changelog-catalog.html',
      'src/catalog/landing_catalog.html' ] },
  Promise {
    [ [Promise],
      [Promise],
      'src/cert-manager/changelog-cert-manager.html',
      'src/cert-manager/landing_cert-manager.html' ] },
  Promise {
    [ 'src/cloud-pak/cloud_pak_integration.html',
      'src/cloud-pak/mgmt_integration.html',
      'src/cloud-pak/product_welcome_cloud_pak_sample.html',
      'src/cloud-pak/version_welcome_sample.html' ] },
  Promise {
    [ [Promise],
      [Promise],
      [Promise],
      'src/cloudctl/calicoctl.html',
      'src/cloudctl/changelog-cloudctl.html',
      'src/cloudctl/cli_case_commands.html',
      'src/cloudctl/cli_catalog_commands.html',
      'src/cloudctl/cli_cm_commands.html',
      'src/cloudctl/cli_commands.html',
      'src/cloudctl/cli_guide.html',
      'src/cloudctl/cli_iam_commands.html',
      'src/cloudctl/cli_iam_create_serviceid.html',
      'src/cloudctl/cli_metering_commands.html',
      'src/cloudctl/cli_pm_commands.html',
      'src/cloudctl/create_helm_cli.html',
      'src/cloudctl/icp_cli.html',
      'src/cloudctl/install_cli.html',
      'src/cloudctl/install_istioctl.html',
      'src/cloudctl/install_kubectl.html',
      'src/cloudctl/landing_cloudctl.html' ] },
  Promise {
    [ [Promise],
      [Promise],
      'src/common-web-ui/changelog-common-web-ui.html',
      'src/common-web-ui/landing_common-web-ui.html' ] },
  Promise {
    [ [Promise],
      'src/configmap-watcher/changelog-configmap-watcher.html' ] },
  'src/do_not_use-summary.html',
  Promise {
    [ [Promise],
      [Promise],
      'src/helm-api/changelog-helm-api.html',
      'src/helm-api/landing_helm-api.html' ] },
  Promise { [ [Promise], 'src/helm-cli/changelog-helm-cli.html' ] },
  Promise {
    [ [Promise],
      [Promise],
      'src/helm-repo/changelog-helm-repo.html',
      'src/helm-repo/landing_helm-repo.html' ] },
  Promise {
    [ [Promise],
      [Promise],
      'src/iam/changelog-iam.html',
      'src/iam/landing_iam.html' ] },
  Promise { [ [Promise] ] },
  Promise {
    [ [Promise],
      [Promise],
      [Promise],
      [Promise],
      [Promise],
      'src/installer/changelog-installer.html',
      'src/installer/landing_installer.html' ] },
  Promise {
    [ 'src/istio-cli/changelog-istio-cli.html',
      'src/istio-cli/install_istioctl.html' ] },
  'src/kc_welcome_cs.html',
  Promise {
    [ 'src/kubectl/changelog-kubectl.html',
      'src/kubectl/install_kubectl.html',
      'src/kubectl/token.html' ] },
  Promise {
    [ [Promise],
      'src/license-service/changelog-license-service.html' ] },
  Promise {
    [ [Promise],
      [Promise],
      'src/logging/changelog-logging.html',
      'src/logging/landing_logging.html' ] },
  Promise { [ [Promise], 'src/mcm-kui/changelog-vwt.html' ] },
  Promise {
    [ [Promise],
      [Promise],
      'src/metering/changelog-metering.html',
      'src/metering/landing_metering.html' ] },
  Promise { [ 'src/mongodb/changelog-mongodb.html' ] },
  Promise {
    [ [Promise],
      [Promise],
      'src/monitoring/changelog-monitoring.html',
      'src/monitoring/landing_monitoring.html' ] },
  Promise {
    [ [Promise],
      [Promise],
      [Promise],
      [Promise],
      [Promise],
      [Promise],
      [Promise] ] },
  Promise { [ [Promise], [Promise] ] },
  Promise { [ [Promise], 'src/support/support.html' ] },
  Promise {
    [ [Promise],
      [Promise],
      'src/system-healthcheck-service/changelog-system-healthcheck-service.html',
      'src/system-healthcheck-service/landing_system-healthcheck-service.html' ] },
  Promise {
    [ 'src/templates/CSV_template.html',
      'src/templates/README_template.html' ] },
  'src/whats_new.html' ]
 ```
 
The call to `getFiles` seems right in the `getInputFileList` method, but wrong in the recursive call to itself in getFiles. This seems to have fixed it, so I will try to submit a pull request asap.

`retVal = retVal.concat(getFiles(f));` => `retVal = retVal.concat(await getFiles(f));`

**Comment by gdelory** [comment]
Created a [PR for this](https://github.com/IBMa/equal-access/pull/172). I already tested locally and it fixed my issue so hopefully it shouldn't be an issue. My test sample is the IBM Cloud common services documentation transformed to HTML using our DCS tools.

**Comment by tombrunet** [comment]
Fixed by #171

**Comment by tombrunet** [comment]
We'll do a deployment later this week.
